src/iOS.Autofill/Utilities/BaseLoginListTableSource.cs

```diff
@@ -31,6 +43,207 @@ public BaseLoginListTableSource(T controller)
 
         protected abstract string LoginAddSegue { get; }
 
+        public bool IsEmpty => Items?.Any() != true;
+
+        public override void RegisterTableViewCells(UITableView tableView)
+        {
+            base.RegisterTableViewCells(tableView);
+
+            tableView.RegisterClassForCellReuse(typeof(ExtendedUITableViewCell), NoDataCellIdentifier);
+        }
+
+        protected override void OnItemsLoaded(string searchFilter, CancellationToken ct)
+        {
+            base.OnItemsLoaded(searchFilter, ct);
+
+            if (Context.IsPreparingListForPasskey && _allowedFido2CipherIds != null)
+            {
+                LoadFido2Ciphers(ct);
+            }
+
+            Controller.OnItemsLoaded(searchFilter);
+        }
+
+        private void LoadFido2Ciphers(CancellationToken ct)
+        {
+            var fido2CiphersToInsert = new List<CipherViewModel>();
+            foreach (var item in Items.Where(i => i?.CipherView?.HasFido2Credential == true))
```

Comment:
❓ Noticed that couple lines down below you were null checking `Items` should it be done here as well?